src/lib/core/trigger.cc

```diff
@@ -50,3 +73,65 @@ trigger_run_reverse(struct rlist *list, void *event)
 			return -1;
 	return 0;
 }
+
+void
+trigger_fiber_run(struct rlist *list, void *event, double timeout)
+{
+	struct trigger *trigger, *tmp;
+	unsigned trigger_count = 0;
+	struct region *region = &fiber()->gc;
+	RegionGuard guard(region);
+	bool expired = false;
+	struct ev_timer timer;
+	ev_timer_init(&timer, trigger_fiber_run_timeout, timeout, 0);
+	timer.data = &expired;
+	ev_timer_start(loop(), &timer);
```

Comment:
I'd add comment noting that we don't check if triggers are timed out during they launch since we want to give them all a chance to run.